sa/sa.go

```diff
@@ -139,10 +139,28 @@ func updateChallenges(authID string, challenges []core.Challenge, tx *gorp.Trans
 			return err
 		}
 		chall.ID = challs[i].ID
-		_, err = tx.Update(chall)
+
+		// We use an explicit UPDATE here with a set of fields we care about instead
+		// of using gorp's `tx.Update()` to avoid the LockCol and concurrent update
```

Comment:
I think in the future, when LockCol doesn't exist, this sentence won't make as much sense. Instead it would be something like "instead of using `tx.Update()` because we are only modifying one column."

Reading through this again, I realize there's another reason we don't need a LockCol on challenges: We only modify the challenges within a transaction. In other words, we were using opportunistic locking at the same time as regular locking! This means that the existing `tx.Update(chall)` is safe enough even without the column-specific UPDATE here. I think I would opt to keep it the same, though I'm open to thoughts otherwise!

Sorry I gave you bad advice earlier about the column-specific update; I hadn't checked this code, and had forgotten that we hold a transaction while we do it.

Reply:
> Reading through this again, I realize there's another reason we don't need a LockCol on challenges: We only modify the challenges within a transaction.

Hmm. That's an interesting point but leaves me a little bit confused. How would the opportunistic lock col error be happening if the concurrent UPDATES are inside of a transaction? Wouldn't the isolation properties result in each incrementing the LockCol appropriately after waiting for the transaction in-flight to complete? 

> I think I would opt to keep it the same, though I'm open to thoughts otherwise!

"the same" -> As an explicit `UPDATE`? Or "the same" -> the same as it is in master, a `tx.Update()`?

> Sorry I gave you bad advice earlier about the column-specific update; I hadn't checked this code, and had forgotten that we hold a transaction while we do it.

No worries. I should have thought about it a little bit more deeply when I was looking at the transactions.